apps/browser/src/vault/fido2/background/fido2.background.ts

```diff
@@ -0,0 +1,352 @@
+import { LogService } from "@bitwarden/common/platform/abstractions/log.service";
+import {
+  AssertCredentialParams,
+  AssertCredentialResult,
+  CreateCredentialParams,
+  CreateCredentialResult,
+  Fido2ClientService,
+} from "@bitwarden/common/vault/abstractions/fido2/fido2-client.service.abstraction";
+import { VaultSettingsService } from "@bitwarden/common/vault/abstractions/vault-settings/vault-settings.service";
+
+import { BrowserApi } from "../../../platform/browser/browser-api";
+import { AbortManager } from "../../background/abort-manager";
+import { Fido2ContentScript, Fido2ContentScriptId } from "../enums/fido2-content-script.enum";
+import { Fido2PortName } from "../enums/fido2-port-name.enum";
+
+import {
+  Fido2Background as Fido2BackgroundInterface,
+  Fido2BackgroundExtensionMessageHandlers,
+  Fido2ExtensionMessage,
+  SharedFido2ScriptInjectionDetails,
+  SharedFido2ScriptRegistrationOptions,
+} from "./abstractions/fido2.background";
+
+export class Fido2Background implements Fido2BackgroundInterface {
+  private abortManager = new AbortManager();
+  private fido2ContentScriptPortsSet = new Set<chrome.runtime.Port>();
+  private currentEnablePasskeysSetting: boolean;
+  private registeredContentScripts: browser.contentScripts.RegisteredContentScript;
+  private readonly sharedInjectionDetails: SharedFido2ScriptInjectionDetails = {
+    allFrames: true,
+    runAt: "document_start",
+  };
+  private readonly sharedRegistrationOptions: SharedFido2ScriptRegistrationOptions = {
+    matches: ["https://*/*"],
+    excludeMatches: ["https://*/*.xml*"],
+    ...this.sharedInjectionDetails,
+  };
+  private readonly extensionMessageHandlers: Fido2BackgroundExtensionMessageHandlers = {
+    fido2AbortRequest: ({ message }) => this.abortRequest(message),
+    fido2RegisterCredentialRequest: ({ message, sender }) =>
+      this.registerCredentialRequest(message, sender),
+    fido2GetCredentialRequest: ({ message, sender }) => this.getCredentialRequest(message, sender),
+  };
+
+  constructor(
+    private logService: LogService,
+    private fido2ClientService: Fido2ClientService,
+    private vaultSettingsService: VaultSettingsService,
+  ) {}
+
+  /**
+   * Initializes the FIDO2 background service. Sets up the extension message
+   * and port listeners. Subscribes to the enablePasskeys$ observable to
+   * handle passkey enable/disable events.
+   */
+  init() {
+    BrowserApi.messageListener("fido2.background", this.handleExtensionMessage);
+    BrowserApi.addListener(chrome.runtime.onConnect, this.handleInjectedScriptPortConnection);
+    this.vaultSettingsService.enablePasskeys$.subscribe(this.handleEnablePasskeysUpdate.bind(this));
```

Comment:
🎨 suggestion: We might be able to remove `currentEnablePasskeySetting` by using `pairwise`

```suggestion
    this.vaultSettingsService.enablePasskeys$.pipe(
        startWith(undefined),
        pairwise()
    ).subscribe((previous, current) => this.handleEnablePasskeysUpdate(previous, new));
```

where `previous` would take the role of `currentEnablePasskeysSetting`

Reply:
Glad I could help! 😁